miio/gateway/sub_device/__init__.py

```diff
@@ -0,0 +1,45 @@
+"""Xiaomi Gateway subdevice base class."""
+
+# flake8: noqa
+from .button import AqaraSquareButton, AqaraSquareButtonV3, AqaraSwitch, Switch
+from .cube import Cube, CubeV2
+from .curtain import Curtain, CurtainB1, CurtainV1
+from .door_sensor import AqaraMagnet, Magnet
+from .light import (
+    AqaraSmartBulbE27,
+    IkeaBulb82,
+    IkeaBulb83,
+    IkeaBulb84,
+    IkeaBulb85,
+    IkeaBulb86,
+    IkeaBulb87,
+    IkeaBulb88,
```

Comment:
I'm wondering how different all of these bulbs really are, i.e., could it be possible to have a single class which exposes the features specific to the bulb in question?

For the time being, I think it's fine to have it as it is as there are so many unknowns at the moment, but at some point it would greatly reduce the boilerplate code if devices could simply be defined in terms of their functionalities (maybe even in simple ini/json/yaml files)? Some food for thought for some later time..

Reply:
They already inherit form one base class that defines everything of these bulbs except the model and zigbee ID.
I do like keeping those since they will probably be needed to match devices using diffrent discovery methods such as cloud that give zigbee_id and not the ID_number that we currently use for matching